staging/src/k8s.io/code-generator/cmd/validation-gen/validators/each.go

```diff
@@ -275,33 +281,58 @@ func (evtv eachValTagValidator) getListValidations(fldPath *field.Path, t *types
 	result := Validations{}
 	result.OpaqueValType = validations.OpaqueType
 
-	var listMap *listMap
+	var listMetadata *listMetadata
 	if lm, found := evtv.byFieldPath[fldPath.String()]; found {
-		if !lm.declaredAsMap {
-			return Validations{}, fmt.Errorf("found listMapKey without listType=map")
+		if lm.declaredAsSet && lm.declaredAsMap {
+			return Validations{}, fmt.Errorf("listType cannot be both et and map")
 		}
-		if len(lm.keyFields) == 0 {
+		if lm.declaredAsMap && len(lm.keyFields) == 0 {
 			return Validations{}, fmt.Errorf("found listType=map without listMapKey")
 		}
-		listMap = lm
+		if len(lm.keyFields) > 0 && !lm.declaredAsMap {
+			return Validations{}, fmt.Errorf("found listMapKey without listType=map")
+		}
+		listMetadata = lm
```

Comment:
We should also check `!lm.declaredAsSet && !lm.declaredAsMap` with an error like "found list metadata without a listType" -- after the other cases, so "listMapKey without listType=map" takes priority (and a comment to that effect)